README.md

```diff
@@ -2,91 +2,111 @@ JavaScript Coding Standards
-} finally {
-  // Statements...
-}
-
-```
+- Use **UPPERCASE_WITH_UNDERSCORES** to name constants. You can define constants in JS with assigning a [primitive value](https://developer.mozilla.org/pl/docs/Glossary/Primitive) with `const`.
+
+  Following examples have one thing in common: they can’t be reassigned nor mutated. This makes them constants.
+
+  ```js
+  const MARGIN_TOP = 2;
+  const DEBUG_MODE = false;
+  const API_URL = 'http://my-api.com';
+  ```
+
+- Avoid UPPERCASE_NAMING for mutable values like arrays or object literals.
+
+  Following examples aren’t constants as they can be easily updated. This is why using uppercase naming is misleading when used with them.
+  
+  ```js
+  const MY_ARR = [1, 2, 3];
+  MY_ARR.splice(0);
+  console.log(MY_ARR);
+  // prints []
+  // Holy moly the array is now empty!
+
+  ```
+  
+  ```
+  const MY_OBJECT = { a: 'something important' };
+  delete MY_OBJECT.a;
+  console.log(MY_OBJECT);
+  // prints {}
+  // Empty!
+  ```
+- jQuery objects should be prefixed with $: `const $items = $('.some-selector');`
+
+When working with existing project it is important to recognize and follow the same naming pattern. No matter what conventions are followed, all names should be **descriptive**, identifying what the data variable holds or what the function does.
 
 ## Comments
 
-Inline documentation for source files should follow the [JSDoc](http://en.wikipedia.org/wiki/JSDoc) formatting conventions. In addition, please use comments to explain the logic behind more tricky parts of code.
-
-## Splitting functionality
-
-When writing a code try to modularize it as much as possible by creating methods/functions. If a project has more JavaScript functionality, add namespaces or even separate it into different files.
+**Avoid obvious comments.** They bring _no value_ and make the code _harder to maintain_. Comment only exceptions and code which doesn’t seem to be clear to understand. The latter can be hard to identify. It’s the code _you_ have written  after all so it’s more obvious to you. Don’t hesitate to ask your peers or developers you consider more experienced to take a look at your code!
+
+## Modules/dependencies
+
+- Use [import](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import) and [export](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/export) to split your code into components.
+- Make sure you import _just what you need_ and not the whole library. Here's [Lodash example](https://stackoverflow.com/a/35251059). It can vary depending on the package you want to use (sometimes can be even impossible!). In general, verify if your JS bundle size isn't suspiciously big.
+- Use [Yarn](https://yarnpkg.com/lang/en/) to install packages. Not only it's fast but also creates a lockfile which make much easier to setup the project for other developers.
+
+## Code structure
+- Prefer to use a class syntax over object literals when structuring your code.
```

Comment:
I like this part but would be great if you could mention how this relates to functional programming because that's what you hear nowadays everywhere. I don't mean we have to strictly follow FP principles in our guidelines, it's just that for someone (me included) this can confusing - is using class ok in terms of FP if I split functionality to small methods?

Reply:
Another thing which would be nice is to provide some real example of well written class and also how to use it in app.js. I remember Michal wrote some nice classes for Getfives, eg.:

```js
import { forEach } from 'iterall';

class Tabs {
  constructor() {
    this.el = document.querySelector('.js-tabs');
    this.tabContentContainers = this.el ? this.el.querySelectorAll('.js-tab') : null;
    this.tabNavigationLinks = this.el ? this.el.querySelectorAll('.js-tabs-nav') : null;

    this.state = {
      activeIndex: 0,
      initCalled: false,
    };

    this.init();
  }

  init() {
    if (this.el === null) {
      return;
    }

    if (!this.state.initCalled) {
      this.setState({ initCalled: true });

      forEach(this.tabContentContainers, (item, i) => this.hideTabs(item, i));
      forEach(this.tabNavigationLinks, (item, i) => this.handleClick(item, i));
      this.tabNavigationLinks[this.state.activeIndex].classList.add('is-active');
    }
  }

  handleClick(link, index) {
    link.addEventListener('click', event => {
      event.preventDefault();
      this.goToTab(index);
    });
  }

  goToTab(index) {
    const { activeIndex } = this.state;
    if (this.isValidIndex(index)) {
      this.tabNavigationLinks[activeIndex].classList.remove('is-active');
      this.tabNavigationLinks[index].classList.add('is-active');
      this.tabContentContainers[activeIndex].classList.add('is-hidden');
      this.tabContentContainers[index].classList.remove('is-hidden');
      this.setState({ activeIndex: index });
    }
  }

  hideTabs(tab, index) {
    const { activeIndex } = this.state;
    if (index !== activeIndex) {
      this.tabContentContainers[index].classList.add('is-hidden');
    }
  }

  setState(newState) {
    this.state = Object.assign({}, this.state, newState);
  }

  isValidIndex(index) {
    return index !== this.state.activeIndex && index >= 0 && index <= this.tabNavigationLinks.length
  }
}

export default Tabs;
```

then in app.js

```js
import Tabs from './Tabs';
new Tabs();
```

Maybe we could use it here, if that's in line with the standards?

Reply:
Hi Lubos,

it doesn't apply to FP. The point was to use class syntax over methods in object literals. Just that!

I like idea with including code examples. 👍

Reply:
@luboskmetko added an example inside "examples" directory. Decided to use native module support so to run the example it's enough to:

1. Install dependencies inside examples directory. Optional, but can enable our Eslint/Prettier settings.
2. Run a server (you can use apache or anything else) and just open an index file inside `class-syntax` directory.

I guess we'll have to update the README with instructions.

Reply:
FYI – I just realised I forgot to push those examples. Duh. Will do it tomorrow as I'm on a laptop now (and was working on my desktop on that).

Reply:
Pushed. ;)